dd-sdk-android/src/main/java/datadog/opentracing/PendingTrace.java

```diff
@@ -14,13 +14,13 @@
 import java.util.List;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.LinkedBlockingDeque;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicReference;
 
-public class PendingTrace extends ConcurrentLinkedDeque<DDSpan> {
+public class PendingTrace extends LinkedBlockingDeque<DDSpan> {
```

Comment:
hmmm...this will block the thread if theres no spans inside...are we sure it is not going to bring any harm ?

Reply:
True